mono/utils/mono-coop-mutex.h

```diff
@@ -130,4 +133,78 @@ mono_coop_cond_broadcast (MonoCoopCond *cond)
 	MONO_EXIT_GC_SAFE;
 }
 
+#else /* DISABLE_THREADS */
+
+struct _MonoCoopMutex {
```

Comment:
Do you need to change the coop ones?  Could just define `MONO_{ENTER,EXIT}_GC_*` macros to be no-ops and leave the existing implementations.